packages/server-core-integration/src/lib/gatewaySetup.ts

```diff
@@ -0,0 +1,215 @@
+const myLogFormat = Winston.format.combine(
+	// Winston.format.errors({ stack: true }),
+	// Winston.format.metadata(),
+	// Winston.format.json()
+	// Winston.format.simple()
+	printf((obj): string => {
+		if (obj instanceof Error) return obj.stack || obj.toString()
+		return JSON.stringify(obj, JSONStringifyCircular())
+	})
+)
+
+/**
+ * Intended to be run first in the Gateways.
+ * Sets up logging, error handling etc.
+ */
+export function setupGatewayProcess(options: { logPath: string; logLevel: string | undefined }) {
+	let logger: Winston.Logger
+	if (options.logPath) {
+		// Log json to file, human-readable to console
+		const transportConsole = new Winston.transports.Console({
+			level: options.logLevel || 'silly',
+			handleExceptions: true,
+			handleRejections: true,
+		})
+		const transportFile = new Winston.transports.File({
+			level: options.logLevel || 'silly',
+			handleExceptions: true,
+			handleRejections: true,
+			filename: options.logPath,
+			format: myLogFormat,
+		})
+
+		logger = Winston.createLogger({
+			transports: [transportConsole, transportFile],
+		})
+		logger.info('Logging to', options.logPath)
+
+		// Hijack console.log:
+		const orgConsoleLog = console.log
+		console.log = function (...args: any[]) {
+			// orgConsoleLog('a')
+			if (args.length >= 1) {
+				try {
+					// eslint-disable-next-line @typescript-eslint/ban-ts-comment
+					// @ts-ignore one or more arguments
+					logger.debug(...args)
+					// logger.debug(...args.map(JSONStringifyCircular()))
+				} catch (e) {
+					orgConsoleLog('CATCH')
+					orgConsoleLog(...args)
+					throw e
+				}
+				orgConsoleLog(...args)
+			}
+		}
+	} else {
+		// custom json stringifier
+
+		// Log json to console
+		const transportConsole = new Winston.transports.Console({
+			level: options.logLevel || 'silly',
+			handleExceptions: true,
+			handleRejections: true,
+			format: myLogFormat,
+		})
+
+		logger = Winston.createLogger({
+			transports: [transportConsole],
+		})
+		logger.info('Logging to Console')
+
+		// Hijack console.log:
+		console.log = (...args: any[]) => {
+			if (args.length >= 1) {
+				// eslint-disable-next-line @typescript-eslint/ban-ts-comment
+				// @ts-ignore one or more arguments
+				logger.debug(...args)
+			}
+		}
+		console.error = (...args: any[]) => {
+			if (args.length >= 1) {
+				// eslint-disable-next-line @typescript-eslint/ban-ts-comment
+				// @ts-ignore one or more arguments
+				logger.error(...args)
+			}
+		}
+	}
+
+	// Because the default NodeJS-handler sucks and wont display error properly
+	process.on('warning', (e: any) => {
+		logger.warn('Unhandled warning, see below')
+		logger.error('error', e)
+		logger.error('error.reason', e.reason || e.message)
+		logger.error('error.stack', e.stack)
+	})
+	// Override the logger methods, to handle a few common cases:
+	for (const method of ['error', 'warn', 'help', 'data', 'info', 'debug', 'prompt', 'http', 'verbose', 'input', 'silly']) {
```

Comment:
What is this block trying to solve? It looks new, and I dont know why we need to apply such a complex hack to these methods.
Is this something we could/should be doing as part of a winston formatter rather than overriding methods?